data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -70,31 +109,61 @@ void viewlist()  // function to display values
     }
 }
 //////////////////////////////////////////////////////////////////////
-
+void test(){
+    insert(1,39);
+    assert(start->info==39);
+    insert(2,10);
+    insert(3,11);
+    deletion(1);
+   assert(start->info!=39);
+    viewlist;
+}
 int main()
 {
-    int n;
-    while (1)
+    int n=0,pos=0,p=0,num=0,c=0;
```

Comment:
I think we just need the self test, there is no need for an interactive mode as we mentioned earlier because that has little to no meaning since that adds no benefit other than side effects.

Reply:
Should probably be optional and ask the user for it. Do you think we should start using these methods now or?

Reply:
I didn't quite understand your point here, could you elaborate?

Reply:
If a user wants to add interactive mode, that's fine. If they don't want to add it, fine as well. Should we start doing this now or?